flow-tests/test-frontend/vite-embedded-session-resync/src/test/java/com/vaadin/viteapp/BasicComponentIT.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2000-2022 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.viteapp;
+
+import java.io.File;
+
+import jakarta.servlet.http.HttpSessionEvent;
+import jakarta.servlet.http.HttpSessionListener;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.webapp.WebAppContext;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.StaleElementReferenceException;
+
+import com.vaadin.flow.testutil.ChromeDeviceTest;
+
+public class BasicComponentIT extends ChromeDeviceTest {
+
+    private Server server;
+
+    private WebAppContext context;
+
+    private String sessionId;
+
+    @Before
+    public void init() throws Exception {
+        setup(8888);
+        getDriver().get(getRootURL());
+        waitForDevServer();
+        getDriver().get(getRootURL() + "/basic-component.html");
+    }
+
+    @Test
+    public void session_resynced_webcomponent_is_active() throws Exception {
+
+        // check if web component works
+        clickButton();
+        Assert.assertEquals("Authentication failure",
+                getAuthenticationResult());
+
+        // simulate expired session by invalidating current session
+        context.getSessionHandler().getSession(sessionId).invalidate();
+
+        // init request to resynchronize expired session and recreate components
+        clickButton();
+
+        try {
+            // it seems WebDriver needs also sync to new session
+            setUsername("");
+        } catch (StaleElementReferenceException ex) {
+            // NOP
+        }
```

Comment:
I would replace this one with a `waitUntil` to avoid the try/catch
```suggestion
        // Wait for embedded component refresh
        waitUntil(d -> $("login-form").first().$("button").first());
```

Reply:
I've tested suggestion and it does not resolve the problem - component is physically present so `waitUntil` goes forward and then on first interaction with component `StaleElementReference` is still thrown.

Reply:
One minor update would be renaming test from `vite-embedded-session-resync` to `vite-embedded-webcomponent-resync`.

Reply:
> I've tested suggestion and it does not resolve the problem - component is physically present so waitUntil goes forward and then on first interaction with component StaleElementReference is still thrown.

Weird :thinking: 
I tested it locally, and it worked for me, but perhaps it is a timing question. So let's use try/catch since it seems to work seamlessly

Reply:
> One minor update would be renaming test from vite-embedded-session-resync to vite-embedded-webcomponent-resync

It seems like a good idea